src/repr/src/adt/datetime.rs

```diff
@@ -1493,8 +1502,12 @@ fn fill_pdt_from_tokens<'a, E: IntoIterator<Item = &'a TimeStrToken>>(
                     )
                 }
                 None => {
+                    // create signed copy of *val
+                    let sval = i64::try_from(i128::from(*val) * i128::from(sign))
+                        .map_err(|_| format!("Unable to parse value {} as a number: number too large to fit in target type", *val))?;
+
                     unit_buf = Some(DateTimeFieldValue {
-                        unit: *val * sign,
+                        unit: sval,
                         fraction: 0,
```

Comment:
Just flagging this because you are not aware, feel free to ignore: Rust supports a thing called [field init shorthands](https://doc.rust-lang.org/book/ch05-01-defining-structs.html#using-the-field-init-shorthand). So this could also be written as:
```rust
let unit = i64::try_from(...).map_err(...)?;
unit_but = Some(DateTimeFieldValue {
    unit,
    fraction: 0,
});
```